From: Alexandros Frantzis <alexandros.frantzis@collabora.com>
Date: Mon, 13 Jul 2020 11:16:28 +0300
Subject: [PATCH] drm: Don't prefer RGBA component widths larger than 8

Don't prefer RGBA component widths larger than 8, since they are
unlikely to be what the users want or properly supported for display.
Such widths can still be used, but only if explicitly requested.

Fixes #125
---
 src/gl-visual-config.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/gl-visual-config.cpp b/src/gl-visual-config.cpp
index dbd87d2a3c3c..013cc646356b 100644
--- a/src/gl-visual-config.cpp
+++ b/src/gl-visual-config.cpp
@@ -108,6 +108,14 @@ GLVisualConfig::score_component(int component, int target, int scale) const
         /* Reward exact matches with the maximum per component score */
         score = MAXIMUM_COMPONENT_SCORE;
     }
+    else if (component > 8 && target <= 8 && scale > 1)
+    {
+        /* Penalize RGBA component widths larger than 8, since they are
+         * unlikely to be what the users want or properly supported for
+         * display. Such widths can still be used, but only if explicitly
+         * requested. */
+        score = UNACCEPTABLE_COMPONENT_PENALTY;
+    }
     else
     {
         /* 
